src/mapblock.cpp

```diff
@@ -377,12 +429,13 @@ void MapBlock::serialize(std::ostream &os_compressed, u8 version, bool disk, int
 	const u8 params_width = 2;
 	if(disk)
 	{
-		MapNode *tmp_nodes = new MapNode[nodecount];
-		memcpy(tmp_nodes, data, nodecount * sizeof(MapNode));
-		getBlockNodeIdMapping(&nimap, tmp_nodes, m_gamedef->ndef());
+		const int size = m_is_mono_block ? 1 : nodecount;
+		MapNode *tmp_nodes = new MapNode[size];
```

Comment:
fwiw if you're editing this anyway you could change this to be `std::unique_ptr<MapBlock[]>`

Reply:
I now think a separate PR might be good. Will do as soon as we merge this one.